connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -380,6 +387,110 @@ protected Map<String, ConfigValue> validateSourceConnectorConfig(SourceConnector
         return configDef.validateAll(config);
     }
 
+    private <T> ConfigInfos validateConverterConfig(
+            Map<String, String> connectorConfig,
+            ConfigValue converterConfigValue,
+            Class<T> converterInterface,
+            Function<T, ConfigDef> configDefAccessor,
+            String converterName,
+            String converterProperty,
+            ConverterType converterType
```

Comment:
I believe this is the only thing which makes this a converter-specific function. All of the other logic appears generic across all plugin types. It could be replaced with a `Consumer<Map<String, String>>` function which mutates the config before it is passed to the plugin, or a `Function<Map<String, String>, Map<String, String>>` if you wanted to be pure.

Then all of the variable names can just become `plugin` instead of `converter`.

Reply:
I like leaving room for a potential future refactor that uses this method for more than just converters, but there is one more part here that's converter-specific: we permit converters to return null `ConfigDef` objects from their `config` methods. We don't allow, e.g., SMTs or predicates to do the same.

To save time on potential future refactoring, I've renamed every parameter and variable in this method to use "plugin" instead of "converter, but I've kept the method name itself (`validateConverterConfig`) the same, and have retained some converter-specific language in log messages. I've also used a `Map<String, String> defaultProperties` instead of the functional programming approach in order to be a little more concise with existing caller code; hope this isn't too controversial. Let me know if this strikes the right balance!

I've also added a Javadoc to this method with examples for most parameters, since the parameter names only helped so much and even I was having trouble recalling how this method worked after a few months without reading it.